client/src/components/CreatePost.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useState } from "react";
+import CssBaseline from "@mui/material/CssBaseline";
+
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import Container from "@mui/material/Container";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Divider from "@mui/material/Divider";
+import Typography from "@mui/material/Typography";
+
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import DialogActions from "@mui/material/DialogActions";
+
+import PreviewPopUp from "./PreviewPopUp";
+import { DialogTitle } from "@mui/material";
+
+const theme = createTheme();
+
+function CreatePost() {
+  // create hooks for input fields
+  const [title, setTitle] = useState("");
+  const [body, setBody] = useState("");
+  const [file, setFile] = useState("");
+  const [tags, setTags] = useState("");
+  const [eventCapacity, setEventCapacity] = useState("");
+  const [eventLocation, setEventLocation] = useState("");
+
+  const [openPopup, setOpenPopup] = useState(false);
+  const [openPopupError, setOpenPopupError] = useState(false);
```

Comment:
We can use an error alert/snackbar instead of an error popup, or just do nothing and highlight that fields are missing